File: Rust/0001-1000/001-100/01-10/01.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut h:HashMap<i32, usize> = HashMap::new();
        let mut v = Vec::new();
        for i in 0..nums.len() {
            if !h.contains_key(&nums[i]) {
                h.insert(target - nums[i], i);
            }
            else {
                v.push(i as i32);
                v.push( *h.get(&nums[i]).unwrap() as i32 );
            }
        }
        v
    }
}

File: Python/0001-1000/001-100/01-10/01.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ret = []
        d = {}
        for i, j in enumerate(nums, 0):
            if j in d.keys():
                ret.append(i)
                ret.append(d[j])
                return ret
            else:
                d[target - j] = i
        return ret
